src/ecc/codes/quantumreedmuller.jl

```diff
@@ -0,0 +1,65 @@
+"""The family of quantum Reed-Muller codes, as discovered by Steane in his 1999 paper [steane1999quantum](@cite).
+
+You might be interested in consulting [zhang1997quantum](@cite), [quan2018fault](@cite) and [campbell2012magic](@cite) as well.
+
+The ECC Zoo has an [entry for this family](https://errorcorrectionzoo.org/c/quantum_reed_muller)
+"""
+struct QuantumReedMuller <: AbstractECC
+    t::Int  
+    r::Int  
+    function QuantumReedMuller(t, r)
+        if t == 1 && r == 2 
+            throw(ArgumentError("Invalid parameters: t = 1, r = 2 is not quantum code as it does not satisfy Hx.Hz' + Hz.Hx' = 0 [steane1999quantum](@cite)."))
```

Comment:
```suggestion
            throw(ArgumentError("Invalid parameters when attempting to construct a quantum Reed-Muller code. t = 1, r = 2 is not quantum code as it does not satisfy commutativity requirements."))
```

The citation would not render in an error message so I just removed it.